blockbase/inc/rest-api.php

```diff
@@ -29,4 +29,38 @@ function blockbase_remove_style_variations_from_child_themes( $response, $handle
 	return $response;
 }
 
-add_filter( 'rest_request_before_callbacks', 'blockbase_remove_style_variations_from_child_themes', 10, 2 );
+add_filter( 'rest_request_before_callbacks', 'blockbase_remove_all_style_variations_from_child_themes', 10, 2 );
+
+/**
+ * Removes any specific style variation from Blockbase child theme, via theme.json.
+ *
+ * @param WP_REST_Response|WP_HTTP_Response|WP_Error|mixed $response Result to send to the client.
+ * @param array                                            $handler  Route handler used for the request.
+ * @param WP_REST_Request                                  $request  Request used to generate the response.
+ *
+ * @return mixed
+ */
+function blockbase_remove_style_variations_from_child_themes( $response, $handler, $request ) {
+	if ( ! isset( $handler['callback'] ) || ! is_array( $handler['callback'] ) ) {
+		return $response;
+	}
+
+	$handler_class    = isset( $handler['callback'][0] ) ? $handler['callback'][0] : null;
+	$handler_method   = isset( $handler['callback'][1] ) ? $handler['callback'][1] : null;
+	$remove_variations = wp_get_global_settings( array( 'custom', 'excludedParentStyleVariations' ) );
+
+	if ( is_a( $handler_class, 'WP_REST_Global_Styles_Controller_Gutenberg' ) && 'get_theme_items' === $handler_method && ! empty( $remove_variations ) ) {
```

Comment:
this needs to change to allow for when the Gutenberg plugin is active, and not, as per https://github.com/Automattic/themes/pull/8638

Reply:
done via: https://github.com/Automattic/themes/pull/8673/commits/9f436031db4f561f54f2ac135ab5c7c08e964291